src/sentry/preprod/api/endpoints/project_preprod_artifact_download.py

```diff
@@ -69,20 +95,78 @@ def get(self, request: Request, project, head_artifact_id, head_artifact) -> Htt
         except File.DoesNotExist:
             return Response({"error": "Preprod artifact file not found"}, status=404)
 
-        try:
-            fp = file_obj.getfile()
-        except Exception:
-            return Response({"error": "Failed to retrieve preprod artifact file"}, status=500)
-
         # All preprod artifacts are zip files
         filename = f"preprod_artifact_{head_artifact.id}.zip"
-
-        response = FileResponse(
-            fp,
-            content_type="application/octet-stream",
-        )
-
-        response["Content-Length"] = file_obj.size
-        response["Content-Disposition"] = f'attachment; filename="{posixpath.basename(filename)}"'
-
-        return response
+        file_size = file_obj.size
+
+        # Check for Range header
+        range_header = request.META.get("HTTP_RANGE")
```

Comment:
There is some range header support here: https://github.com/getsentry/sentry/blob/12fe967304e41448c6dbd19e42da263200808e6f/src/sentry/replays/endpoints/project_replay_video_details.py#L85 I don't if any of it is usable for is.

Reply:
Looking at it, I don't think it's needed to follow their more pattern - i think ours is simpler

Reply:
spoke to soon, going to use a different helper method of theirs 😂